server/stream.go

```diff
@@ -3818,7 +3811,7 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 		} else if mset.numFilter > 0 {
 			// Assume no interest and check to disqualify.
 			noInterest = true
-			for _, o := range mset.consumers {
+			for _, o := range mset.cList {
```

Comment:
But even if you are under stream lock, you would need to protect with the new mutex to avoid races with other code that does not use stream lock but uses only the new lock